### ISSUE TITLE
Add `rbenv shell -` to switch back to previous version of Ruby

### ISSUE THREAD

**Issue by tenderlove** [rbenv/rbenv#854]
Hi,

I think it would be cool if you could do `rbenv shell -` and have `-` mean "the previous version I was using".  Similar to `cd -` and `git checkout -`.  I usually use trunk Ruby installed in to my ruby-build directories, but sometimes I need to switch to stable, then switch back.

I'm not sure if this is the right place to file the ticket.  Also, I'll try to implement this eventually but I don't have the time right now. I just wanted to file a ticket so that I won't forget and in case other people have time.

Thanks!

**Comment by mislav** [comment]
Thanks for the suggestion! The implementation should be trivial: the previous value should simply be saved in a temporary environment variable that doesn't have to be exported like RBENV_VERSION is.

However, what would be the result of running `rbenv shell -` when RBENV_VERSION was previously unset before running `rbenv shell 2.2.4`, for instance?

**Comment by tenderlove** [comment]
> However, what would be the result of running rbenv shell - when RBENV_VERSION was previously unset before running rbenv shell 2.2.4, for instance?

I think we could just do something similar to bash:

```
Last login: Mon Jan 11 11:38:10 on ttys011
[aaron@TC eventmachine (master)]$ cd -
-bash: cd: OLDPWD not set
[aaron@TC eventmachine (master)]$
```

**Comment by shakycode** [comment]
:+1:

**Comment by mislav** [comment]
Merged in 2207c27cb4fce9e7221cd4b1c715b95267538334
